src/app/pages/storage/disks/disk-list/disk-list.component.ts

```diff
@@ -127,7 +128,7 @@ export class DiskListComponent {
     this.ws.call('smart.test.disk_choices').subscribe((res) => this.SMARTdiskChoices = res, (err) => new EntityUtils().handleWSError(this, err));
   }
 
-  getActions(parentRow: any) {
+  getActions(parentRow: any): EntityTableAction[] {
     const actions = [{
```

Comment:
Wouldn't it be better to assign the type here in the first place instead of type-casting on return?

Reply:
It wouldn't work, declaring actions as EntityTableAction[] will require all fields within EntityTableAction to be implemented. Casting will silence this issue. 
The underlying problem is that EntityTableAction is not correctly defined, but this is outside of the scope of this PR.